project4/name_server/nameserver_main.cpp

```diff
@@ -0,0 +1,48 @@
+/**
+ * @file Main executable for Name Server
+ */
+
+#include <loguru.hpp>
+
+#include <iostream>
+
+#include "nameserver_driver.h"
+#include "bootstrap_driver.h"
+
+void start_nameserver(char* config) {
```

Comment:
If you take my suggestion about unifying `BootstrapDriver` and `NameserverDriver`, some of this code can be simplified.